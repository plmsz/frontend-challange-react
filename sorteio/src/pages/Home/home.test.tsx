import { Home } from './index'
import { act, fireEvent, render, screen, waitFor } from '../../tests/renderWithContext'
import userEvent from '@testing-library/user-event'

it('renders correctly', async () => {
  const { asFragment } = render(<Home />)

  const title = await screen.findByRole('heading', { name: /mega-sena/i })
  expect(title).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot('create')
})

it('should show the information of the first lottery such as the title, contest number, the date and the results', async () => {
  render(<Home />)
  await act(async () => Promise.resolve())

  const megaOption = screen.getByRole('option', { name: /mega/i }) as HTMLOptionElement
  expect(megaOption.selected).toBe(true)

  const title = screen.getByRole('heading', { name: /mega/i })
  expect(title).toBeInTheDocument()

  const contextNumber = screen.getByRole('heading', { name: /1000/i, exact: false })
  expect(contextNumber).toBeInTheDocument()

  const listItems = screen.getAllByRole('listitem')
  const itemsContent = listItems.map((item) => item.textContent)

  expect(itemsContent).toEqual(['1', '2', '3', '4', '5', '6'])
})
it('should show the a title, contest number, the date and the results, after change the lottery', async () => {
  render(<Home />)
  await act(async () => Promise.resolve())

  const combobox = screen.getByRole('combobox', { name: /loterias/i })

  const lotofacilOption = screen.getByRole('option', { name: /lotofácil/i }) as HTMLOptionElement

  userEvent.selectOptions(combobox, lotofacilOption)
  expect(lotofacilOption.selected).toBe(true)

  const title = await screen.findByRole('heading', { name: /lotofácil/i })
  expect(title).toBeInTheDocument()

  const contextNumber = screen.getByRole('heading', { name: /3000/i, exact: false })
  expect(contextNumber).toBeInTheDocument()

  const listItems = screen.getAllByRole('listitem')
  const itemsContent = listItems.map((item) => item.textContent)

  expect(itemsContent).toEqual(['100', '200', '300', '400', '500', '600', '700', '800', '900', '1000', '2000'])
})
it('should render a button at the bottom of the page if the screen is long enough', async () => {
  render(<Home />)
  await act(async () => Promise.resolve())

  fireEvent.scroll(window, { target: { scrollY: 180 } })
  const img = screen.getByRole('button', { name: /voltar para o topo da página/i })

  expect(img).toBeInTheDocument()
})
