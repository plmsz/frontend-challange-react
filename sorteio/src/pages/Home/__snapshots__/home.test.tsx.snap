// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: create 1`] = `
<DocumentFragment>
  <div
    class="sc-ftvSup KoSIc"
  >
    <header
      class="sc-jSMfEi boQBfG"
    >
      <div
        class="sc-bczRLJ ePhNbS"
      >
        <select
          aria-label="Loterias"
          class="sc-eCYdqJ bBEFDc"
          name="lotteryDropdown"
        >
          <option
            value="mega-sena"
          >
            mega-sena
          </option>
          <option
            value="quina"
          >
            quina
          </option>
          <option
            value="lotofácil"
          >
            lotofácil
          </option>
          <option
            value="lotomania"
          >
            lotomania
          </option>
          <option
            value="timemania"
          >
            timemania
          </option>
          <option
            value="dia de sorte"
          >
            dia de sorte
          </option>
        </select>
        <div
          class="sc-gsnTZi gEzBOl"
        >
          <picture>
            <source
              media="(max-width: 768px)"
              srcset="logoPrimary.svg"
            />
            <source
              media="(min-width: 768px)"
              srcset="logoSecondary.svg"
            />
            <img
              alt="logo - trevo de 4 folhas"
              src="logoPrimary.svg"
            />
          </picture>
          <h1>
            mega-sena
          </h1>
        </div>
        <h2
          class="titleMobile"
        >
          CONCURSO Nº 1000
        </h2>
        <div
          class="titleDesktop"
        >
          <h2>
            CONCURSO
          </h2>
          <p>
            1000 - 
          </p>
        </div>
      </div>
    </header>
    <div
      class="sc-iBkjds bbYePF"
    >
      <h1
        class="sc-dkzDqf iDJqVp"
        color="quina"
      >
        Aposte, hoje pode ser o seu dia!
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
